rubi/rubi/client.py

```diff
@@ -70,231 +70,224 @@ def __init__(
 
-        base, quote = pair_name.split("/")
+    def get_transaction_receipt(
+        self,
+        transaction_hash: str,
+        pair_names: Optional[List[str]] = None,
+    ) -> TransactionReceipt:
+        """Get the transaction receipt for a given transaction hash.
+
+        :param transaction_hash: The transaction hash.
+        :type transaction_hash: str
+        :param pair_names: If handling a transaction that interacted with the rubicon market we can decode events into
+            human-readable format if we know the pairs in the transaction. (defaults to None and no decoding is done)
+        :type pair_names: Optional[List[str]]
+        :return: A TransactionReceipt for the transaction hash.
+        :rtype: TransactionReceipt
+        :return:
+        """
 
-        base_asset = ERC20.from_network(
-            name=base, network=self.network, wallet=self.wallet, key=self.key
-        )
-        quote_asset = ERC20.from_network(
-            name=quote, network=self.network, wallet=self.wallet, key=self.key
+        transaction_receipt = self.network.transaction_handler.get_transaction_receipt(
+            transaction_hash=transaction_hash,
         )
 
-        current_base_asset_allowance = None
-        current_quote_asset_allowance = None
-
-        if self.wallet is not None and self.key is not None:
-            current_base_asset_allowance = base_asset.to_decimal(
-                number=base_asset.allowance(
-                    owner=self.wallet, spender=self.market.address
-                )
-            )
-            current_quote_asset_allowance = quote_asset.to_decimal(
-                number=quote_asset.allowance(
-                    owner=self.wallet, spender=self.market.address
-                )
-            )
-
-            if current_base_asset_allowance == Decimal(
-                "0"
-            ) or current_quote_asset_allowance == Decimal("0"):
-                logger.warning(
-                    "allowance for base or quote asset is zero. this may cause issues when placing orders"
-                )
-
-        self._pairs[f"{base}/{quote}"] = Pair(
-            name=pair_name,
-            base_asset=base_asset,
-            quote_asset=quote_asset,
-            current_base_asset_allowance=current_base_asset_allowance,
-            current_quote_asset_allowance=current_quote_asset_allowance,
+        return self._handle_transaction_receipt_raw_events(
+            transaction_receipt=transaction_receipt,
+            pair_names=pair_names,
         )
 
-        # only edit allowance if client has signing rights
-        if self.wallet is not None and self.key is not None:
-            self.update_pair_allowance(
-                pair_name=pair_name,
-                new_base_asset_allowance=base_asset_allowance,
-                new_quote_asset_allowance=quote_asset_allowance,
-            )
-
-    def get_pairs_list(self) -> List[str]:
```

Comment:
we may want to keep this one, and have a deprecation warning attached to it. here, we could return all pairs that can be created from `self.network.tokens`